### ISSUE TITLE
All settings gone, plugin incorrect, no connection to DB?

### ISSUE THREAD

**Issue by SphaeroX** [mmone/OctoprintKlipperPlugin#57]
About your environment:
 - Ooctopi
 - Octoprint version 1.3.12

I always had OctoKlipper running without any problems, I also created my own macros which was really handy. But from one day to the next the plugin doesn't work anymore.
The macros are gone, just an empty button, in the settings everything is gone, too. I can't save anything new, it stays empty.

I have already uninstalled the plugin, rebooted Octopi and installed it again, still the same error. All files of OctoKlipper were deleted when I uninstalled it. I think there is something wrong with the database and it is not removed properly.

Where is the database and how can I remove it manually for testing?

![1](https://user-images.githubusercontent.com/6785110/73639784-533b5b80-466d-11ea-9285-4dc80e89d662.JPG)
![2](https://user-images.githubusercontent.com/6785110/73639785-533b5b80-466d-11ea-8f20-3b9cca38b22e.JPG)

[plugin_pluginmanager_console.log](https://github.com/mmone/OctoprintKlipperPlugin/files/4146997/plugin_pluginmanager_console.log)
[octoprint.log](https://github.com/mmone/OctoprintKlipperPlugin/files/4146998/octoprint.log)

**Comment by OleNielsen** [comment]
I had these symptoms too. What worked for me was SSH'ing into the Pi and editing printer.cfg with nano editor. Start by examining every line : tab 'end'-key and verify that there is no 'invisible' character at the end of any textlines. Otherwise delete these.

Secondly, I found two occurences where a character in a #-comment apparently changed the rest of the textline to be interpreted by OctoKlipper as a statement. Nano editor shows statements in white colored text, and comments in cyan colored text; I observed that a commentary line would change textcolor from cyan to white
`# this is a commentary but a character xC3 changed this part to white text.`

As Klipper ignores lines starting with a hash-symbol, Klipper shrugs at this and accepts printer.cfg as is.
OctoKlipper plugin editor on the other hand goes south.
 
Having tidyed up printer.cfg and rebooted the Pi, OctoKlipper worked again.
